038-optimization-proposal-autoapproval.md

```diff
@@ -0,0 +1,66 @@
+# Auto-approval mechanism for optimization proposals
+
+This proposal is about allowing the user to create a `KafkaRebalance` custom resource for getting an optimization proposal and running a rebalance, but without the need of the manual approval via the `strimzi.io/rebalance=approve` annotation.
+It means that the user can just create the `KafkaRebalance` custom resource and the corresponding generated optimization proposal will be automatically approved.
+
+## Current situation
+
+Currently, when the users want to run a cluster rebalance, they have to create a `KafkaRebalance` custom resource for getting an optimization proposal first.
+After getting the proposal, the only way to start the rebalancing operation is by approving it by annotating the custom resource with the `strimzi.io/rebalance=approve` annotation.
+There is the need for a manual intervention of the user and this is a two steps process.
+
+## Motivation
+
+In some cases, the manual approval process of the rebalance proposal is not necessary.
+Automatically approving an optimization proposal can save time from an operational point of view.
+It enables more automation where just creating a `KafkaRebalance` custom resource can go straight to the cluster rebalance without any additional manual interaction.
+
+## Proposal
+
+The `KafkaRebalance` custom resource can be extended with a new `spec.autoApproval` field for this purpose.
+
+### Auto-approval
+
+The auto-approval can be enabled just adding an empty `spec.autoApproval` field.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaRebalance
+metadata:
+  name: my-rebalance
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  mode: # any mode
+  autoApproval: {}
+```
+
+The reason for not having a simple boolean `spec.autoApproval` field is for leaving space for a future extension, maybe adding some criteria or rules that have to be matched for auto-approving the optimization proposal.
```

Comment:
Ooo I like this, I can foresee some kind of pluggable `RebalanceApprover` interface.